src/tribler/core/components/watch_folder/watch_folder.py

```diff
@@ -27,25 +27,18 @@ def __init__(self, state_dir: Path, settings: WatchFolderSettings, download_mana
         self._logger.info(f'Initialised with {settings}')
 
     def start(self):
-        self.register_task("check watch folder", self.check_watch_folder, interval=WATCH_FOLDER_CHECK_INTERVAL)
+        self.register_task("check watch folder", self.check_watch_folder_handle_exceptions,
+                           interval=WATCH_FOLDER_CHECK_INTERVAL)
 
     async def stop(self):
         await self.shutdown_task_manager()
 
-    def cleanup_torrent_file(self, root, name):
```

Comment:
> Can it lead to situation when the download manager will try to start the same corrupted torrent again and again?

Yes, but it will not affect a user. I made this change after discussing with you the previous version of PR. This change is a consequence of a new approach.

Reply:
Fixed